app/scripts/views/components/barChart/barChart.js

```diff
@@ -295,87 +299,123 @@
 
     content.rangeFilter = function(logScaleChecked, _filter) {
       var tempFilters_ = [];
-      var hasBar = false;
       var minNumBarPoint = '';
       var maxNumBarPoint = '';
       var selectedNumBar = [];
       var hasNA = false;
       var hasGreaterOutlier = false;
       var hasSmallerOutlier = false;
       var startNumIndex = 0;
-      var endNumIndex = 0;
+      var endNumIndex = opts_.xDomain.length - 1;
 
       tempFilters_[0] = '';
       tempFilters_[1] = '';
+      
+      // set start and end indexes for number bars
+      if (data_.noGrouping) {
+        if (data_.hasNA) {
+          endNumIndex = opts_.xDomain.length - 2;
```

Comment:
I'm curious since you didn't run into any issue on this, but does it matter to check xDomain length?